libr/bin/format/elf/elf.c

```diff
@@ -2504,27 +2507,32 @@ int Elf_(r_bin_elf_is_big_endian)(ELFOBJ *bin) {
 
 /* XXX Init dt_strtab? */
 char *Elf_(r_bin_elf_get_rpath)(ELFOBJ *bin) {
-	char *ret = NULL;
-	int j;
+	char *ret;
+	Elf_(Xword) val;
 
-	if (!bin || !bin->phdr || !bin->dyn_buf || !bin->strtab) {
+	if (!bin || !bin->phdr || !bin->strtab) {
```

Comment:
Use r_return_val_if_fail for bin at least because it should never benull. Phdr and strtab can be null so its fine to chk it lik ethis